integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java

```diff
@@ -123,19 +123,33 @@ public void updateResource(String resourcePath, String sourcePath, boolean backu
     /**
      * Function is to check if the registry resources are exist in filepath
      *
-     * @param sourcePath path to registry resource
      * @param resourcePath path to resource file path from registry resource
      * @param filename registry filename
      */
 
-    public boolean checkResourceExist(String sourcePath, String resourcePath, String filename) throws Exception {
-        try {
-            String path = sourcePath + resourcePath + filename;
-            return path.contains(filename);
+    public boolean checkResourceExist(String resourcePath, String filename) throws Exception {
+
+        boolean isExists = false;
+
+        StringBuilder pathBuilder = new StringBuilder();
+        pathBuilder.append(System.getProperty("carbon.home")).append(File.separator).append("registry").append(File.separator);
+
+        if (resourcePath.startsWith(CONF_PATH)) {
+            String relativePath = resourcePath.substring(5).replace('/', File.separatorChar);
+            pathBuilder.append("config").append(relativePath);
+        } else if (resourcePath.startsWith(GOV_PATH)) {
+            String relativePath = resourcePath.substring(4).replace('/', File.separatorChar);
+            pathBuilder.append("governance").append(relativePath);
+        } else {
+            throw new MicroRegistryManagerException("Exception occurred while checking the registry resource file: " + resourcePath);
+        }
 
-        } catch (Exception e) {
-            throw new Exception("Exception occurred while checking the registry resource file.", e);
+        if(new File(pathBuilder + filename).isFile()){
```

Comment:
Shall we directly return new File(pathBuilder + filename).isFile() 
Then we can omit this condition check